src/main/java/org/mskcc/cbio/oncokb/service/SamlService.java

```diff
@@ -0,0 +1,99 @@
+package org.mskcc.cbio.oncokb.service;
+
+import java.util.function.Supplier;
+import javax.annotation.PostConstruct;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.mskcc.cbio.oncokb.config.application.ApplicationProperties;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+import software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleWithSamlCredentialsProvider;
+import software.amazon.awssdk.services.sts.model.AssumeRoleWithSamlRequest;
+
+@Service
+public class SamlService {
+    private StsAssumeRoleWithSamlCredentialsProvider credentialsProvider;
+
+    private final String MSK_IDP_URL ="https://ssofed.mskcc.org/idp/startSSO.ping?PartnerSpId=urn:amazon:webservices";
+    private final String MSK_USERNAME_FIELD = "pf.username";
+    private final String MSK_PASSWORD_FIELD = "pf.pass";
+
+    private final ApplicationProperties applicationProperties;
+
+    public SamlService(ApplicationProperties applicationProperties) {
+        this.applicationProperties = applicationProperties;
+    }
+
+    @PostConstruct
+    private void initSecurityTokenServiceProvider() {
+        if (applicationProperties.getSamlAws() != null) {
+            AssumeRoleWithSamlRequest samlRequest = AssumeRoleWithSamlRequest
+                .builder()
+                .principalArn(applicationProperties.getSamlAws().getPrincipalArn())
+                .roleArn(applicationProperties.getSamlAws().getRoleArn())
+                .durationSeconds(28800)
+                .build();
+
+            Supplier<AssumeRoleWithSamlRequest> supplier = () ->
```

Comment:
Will it throw exception when getSamlResponse returns null?

Reply:
An exception will be thrown by the aws security token service saying that the saml assertion cannot be null. For `S3Service.getObject()`, the exception will be caught and we currently return an empty optional.